--- a/apple/internal/watchos_rules.bzl
+++ b/apple/internal/watchos_rules.bzl
@@ -978,7 +978,7 @@
         ),
     ]
 
-    if platform_prerequisites.platform.is_device:
+    if platform_prerequisites.platform.is_device and "disable_legacy_signing" not in features:
         processor_partials.append(
             partials.provisioning_profile_partial(
                 actions = actions,
@@ -1288,7 +1288,7 @@
         ),
     ]
 
-    if platform_prerequisites.platform.is_device:
+    if platform_prerequisites.platform.is_device and "disable_legacy_signing" not in features:
         processor_partials.append(
             partials.provisioning_profile_partial(
                 actions = actions,
@@ -1753,7 +1753,7 @@
         ),
     ]
 
-    if platform_prerequisites.platform.is_device:
+    if platform_prerequisites.platform.is_device and "disable_legacy_signing" not in features:
         processor_partials.append(
             partials.provisioning_profile_partial(
                 actions = actions,
